w3d_translator/w3d_translator.py

```diff
@@ -54,111 +55,127 @@ def create_out(out_dir: Path, force: bool):
     except FileNotFoundError:
         raise ValidationError("Error: OUTPUT directory is not valid")
     except ValidationError as e:
-        err_console.print(e, file=sys.stderr, style="red")
+        err_console.print(e)
         exit(1)
 
 
 # Copy files from the project directory to the Unity output directory
 def copy_files(project_dir: Path, unity_dir: Path, unity_copy: Path):
     try:
-        shutil.copytree(str(STARTER_PROJECT), str(unity_dir))
-        shutil.copytree(
-            project_dir,
-            unity_copy,
+        with console.status("Copying files"):
+            shutil.copytree(str(STARTER_PROJECT), str(unity_dir))
+            shutil.copytree(
+                project_dir,
+                unity_copy,
+            )
+    except (ValidationError, UnityError, Exception) as e:
+        err_console.print(e)
+    else:
+        console.print(
+            f"Copied files from [cyan]{project_dir}/[/cyan]"
+            + f" to [cyan]{unity_dir}\\"
         )
-    except Exception as e:
-        # TODO 55: Catch this exception (skip file)
-        err_console.print(e, file=sys.stderr, style="red")
 
 
-# Translate an XML file using Unity's CLI
-def translate_file(unity_dir: Path, xml_path: Path, filename: str):
-    try:
-        validate_xml(xml_path)
-    except XmlError as e:
-        err_console.print(e, file=sys.stderr, style="red")
+# Translate valid xml files (ignore invalid)
+def translate_files(unity_dir: Path, unity_copy: Path):
+    # TODO 138: Use a live progress group to keep track of the different files
+    # TODO 3: Launch each file in its own process
+    all_good = True
+    for xml_path in unity_copy.rglob("*.xml"):
+        console.print(f"Translating file: '{xml_path.name}'")
+        try:
+            translate_file(unity_dir, xml_path)
+        except (XmlError, UnityError) as e:
+            err_console.print(e)
+            all_good = False
+
+    if all_good:
+        console.print(
+            ":white_check_mark: "
+            + f"[cyan]{unity_dir.name}/ "
+            + "translated successfully[/cyan]\n"
+        )
     else:
-        console.print(f"'{filename}' is valid")
+        console.print(
+            f"[yellow]{unity_dir.name}/ "
+            + "translated with some errors [/yellow]\n"
+        )
```

Comment:
Do you really need the `all_good`  variable?

Reply:
When a specific file throws an exception in Unity the file gets skipped and that scene is not saved. But the python script does not crash. I want a way to tell the user that some of the files could not be translated at the end. Perhaps I could keep a running count of how many fail and how many translate correctly?